### PR TITLE
Release notes for v1.5.0-preview2

### PR DESCRIPTION
**Pull request by harishsk**

### CODE REVIEW

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- Sweep Trimming of Whitespace in AutoML ([#3918](https://github.com/dotnet/machinelearning/pull/3918))
+- Use random file name for AutoML experiment folder ([#4657](https://github.com/dotnet/machinelearning/pull/4657))
+
+
+## **Enhancements**
+- Added in support for System.DateTime type for the DateTimeTransformer ([#4661](https://github.com/dotnet/machinelearning/pull/4661))
+- Additional changes to ExpressionTransformer ([#4614](https://github.com/dotnet/machinelearning/pull/4614))
+
+## **Test updates**
+- **Code analysis updates**
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: We don't ship code analyzer to our nuget users, why is this relevant to release notes?

CC: @eerhardt

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  - Update analyzer test library ([#4740](https://github.com/dotnet/machinelearning/pull/4740))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: We don't ship to users either.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  - Enable the internal code analyzer for test projects ([#4731](https://github.com/dotnet/machinelearning/pull/4731))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: This is not shipped to the nuget users.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  - Implement MSML_ExtendBaseTestClass (Test classes should be derived from BaseTestClass) ([#4746](https://github.com/dotnet/machinelearning/pull/4746))
+  - Enable MSML_TypeParamName for the full solution ([#4762](https://github.com/dotnet/machinelearning/pull/4762))
+- **Better logging from tests**
+  - Ensure tests capture the full log ([#4710](https://github.com/dotnet/machinelearning/pull/4710))
+  - Fix failure to capture test failures ([#4716](https://github.com/dotnet/machinelearning/pull/4716))
+  - Collect crash dump upload dump and pdb to artifact ([#4666](https://github.com/dotnet/machinelearning/pull/4666))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: How is collect crash dump relevant to nuget users? It will almost make them feel their data is being collected.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+- Enable Conditional Numerical Reproducibility for tests ([#4569](https://github.com/dotnet/machinelearning/pull/4569))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: This is specific to test projects, we don't ship test projects.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- Additional changes to ExpressionTransformer ([#4614](https://github.com/dotnet/machinelearning/pull/4614))
+
+## **Test updates**
+- **Code analysis updates**
+  - Update analyzer test library ([#4740](https://github.com/dotnet/machinelearning/pull/4740))
+  - Enable the internal code analyzer for test projects ([#4731](https://github.com/dotnet/machinelearning/pull/4731))
+  - Implement MSML_ExtendBaseTestClass (Test classes should be derived from BaseTestClass) ([#4746](https://github.com/dotnet/machinelearning/pull/4746))
+  - Enable MSML_TypeParamName for the full solution ([#4762](https://github.com/dotnet/machinelearning/pull/4762))
+- **Better logging from tests**
+  - Ensure tests capture the full log ([#4710](https://github.com/dotnet/machinelearning/pull/4710))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: We don't ship tests.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+- ImageClassificationTrainer now checks the existence of input folder before training ([#4691]( https://github.com/dotnet/machinelearning/pull/4691))
+- Sweep Trimming of Whitespace in AutoML ([#3918](https://github.com/dotnet/machinelearning/pull/3918))
+- Use random file name for AutoML experiment folder ([#4657](https://github.com/dotnet/machinelearning/pull/4657))
+
+
+## **Enhancements**
+- Added in support for System.DateTime type for the DateTimeTransformer ([#4661](https://github.com/dotnet/machinelearning/pull/4661))
+- Additional changes to ExpressionTransformer ([#4614](https://github.com/dotnet/machinelearning/pull/4614))
+
+## **Test updates**
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: This whole section is not shipped to the users, why is this included in release notes? Who is your target audience and why would they care about stuff that is not shipped part of nuget?

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- Roadmap update ([#4704](https://github.com/dotnet/machinelearning/pull/4704))
+- Added release process documentation to README.md ([#4402](https://github.com/dotnet/machinelearning/pull/4402))
+- Fix documentation of SvmLightLoader ([#4616](https://github.com/dotnet/machinelearning/pull/4616))
+- Correct KMeans scoring function doc ([#4705](https://github.com/dotnet/machinelearning/pull/4705))
+- Several typo fixes thanks to [@MaherJendoubi](https://github.com/MaherJendoubi) ([#4627](https://github.com/dotnet/machinelearning/pull/4627), [#4631](https://github.com/dotnet/machinelearning/pull/4631), [#4626](https://github.com/dotnet/machinelearning/pull/4626) [#4617](https://github.com/dotnet/machinelearning/pull/4617), [#4633](https://github.com/dotnet/machinelearning/pull/4633), [#4629](https://github.com/dotnet/machinelearning/pull/4629), [#4642](https://github.com/dotnet/machinelearning/pull/4642))
+- Other typo fixes:([#4628](https://github.com/dotnet/machinelearning/pull/4628), [#4685](https://github.com/dotnet/machinelearning/pull/4685))
+
+## **CLI and AutoML API**
+  - CodeGen For AzureAttach ([#4498](https://github.com/dotnet/machinelearning/pull/4498))
```
Comment (LittleLittleCloud) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: We don't want user aware of this change because this is only for ModelBuilder Azure Attach story

Comment (LittleLittleCloud) [reply docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: plase add
 - Using invariance culture when converting to string ( [#4635](https://github.com/dotnet/machinelearning/pull/4635) )
and 
 - Fix NullReferenceException when it comes to Recommendation in AutoML and CodeGenerator ( [#4774](https://github.com/dotnet/machinelearning/pull/4774) )

Thanks

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+- Fix bug in WordBagEstimator when training on empty data ([#4696](https://github.com/dotnet/machinelearning/pull/4696))
+- Added Cancellation mechanism to Image Classification (through the experimental nuget) (fixes #4632) ([#4650](https://github.com/dotnet/machinelearning/pull/4650))
+-	Changed F1 score to return 0 instead of NaN when Precision + Recall is 0 ([#4674](https://github.com/dotnet/machinelearning/pull/4674))
+- TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
+- Changed onnx export to append a .onnx string to column names to support better mapping to ML.NET output column names ([#4734](https://github.com/dotnet/machinelearning/pull/4734))
+-	TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
+- ImageClassificationTrainer now checks the existence of input folder before training ([#4691]( https://github.com/dotnet/machinelearning/pull/4691))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: this is ImagLoading transform not ImageClassification trainer, two very different things, one is a transformer and other is a trainer.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
+# [ML.NET](http://dot.net/ml) 1.5.0-preview2
+
+## **New Features (IN-PREVIEW, please provide feedback)**
+- **TimeSeriesImputer** ([#4623](https://github.com/dotnet/machinelearning/pull/4623))
+- **LDSVM Trainer** ([#4060](https://github.com/dotnet/machinelearning/pull/4060))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: >LDSVM Traine [](start = 4, length = 12)

Describe what this trainer is and where it is used and then provide links to samples.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
+# [ML.NET](http://dot.net/ml) 1.5.0-preview2
+
+## **New Features (IN-PREVIEW, please provide feedback)**
+- **TimeSeriesImputer** ([#4623](https://github.com/dotnet/machinelearning/pull/4623))
```
Comment (codemzs) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: >TimeSeriesImputer [](start = 4, length = 17)

Describe what this transformer is and where it is used and then provide links to samples.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md (hunk 10 above)
Comment (justinormont) [reply docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: Description:
LDSVM uses trees as its SVM kernel to create a non-linear binary learner.

Sample:
https://github.com/dotnet/machinelearning/blob/c819d77e9250c68883713d5f1cd79b8971a11faf/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LdSvm.cs

Reference links:
http://manikvarma.org/pubs/jose13.pdf
http://manikvarma.org/talks/ICML13.pdf

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+- **Export-to-ONNX for below components:**
+    - SlotsDroppingTransformer ([#4562](https://github.com/dotnet/machinelearning/pull/4562))
+    - ColumnSelectingTransformer ([#4590](https://github.com/dotnet/machinelearning/pull/4590))
+    - VectorWhiteningTransformer ([#4577](https://github.com/dotnet/machinelearning/pull/4577))
+    - NaiveBayesMulticlassTrainer ([#4636](https://github.com/dotnet/machinelearning/pull/4636))
+  	- PlattCalibratorTransformer ([#4699](https://github.com/dotnet/machinelearning/pull/4699))
+
+  
+## **Bug Fixes**
+- Fix for OneVersusAllTrainer([#4698](https://github.com/dotnet/machinelearning/pull/4698))
```
Comment (justinormont) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: Space:
```suggestion
- Fix for OneVersusAllTrainer ([#4698](https://github.com/dotnet/machinelearning/pull/4698))
```

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+- Update documentation to stop mentioning interfaces that no longer exist ([#4673](https://github.com/dotnet/machinelearning/pull/4673))
+- Roadmap update ([#4704](https://github.com/dotnet/machinelearning/pull/4704))
+- Added release process documentation to README.md ([#4402](https://github.com/dotnet/machinelearning/pull/4402))
+- Fix documentation of SvmLightLoader ([#4616](https://github.com/dotnet/machinelearning/pull/4616))
+- Correct KMeans scoring function doc ([#4705](https://github.com/dotnet/machinelearning/pull/4705))
+- Several typo fixes thanks to [@MaherJendoubi](https://github.com/MaherJendoubi) ([#4627](https://github.com/dotnet/machinelearning/pull/4627), [#4631](https://github.com/dotnet/machinelearning/pull/4631), [#4626](https://github.com/dotnet/machinelearning/pull/4626) [#4617](https://github.com/dotnet/machinelearning/pull/4617), [#4633](https://github.com/dotnet/machinelearning/pull/4633), [#4629](https://github.com/dotnet/machinelearning/pull/4629), [#4642](https://github.com/dotnet/machinelearning/pull/4642))
+- Other typo fixes:([#4628](https://github.com/dotnet/machinelearning/pull/4628), [#4685](https://github.com/dotnet/machinelearning/pull/4685))
```
Comment (justinormont) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: Space:
```suggestion
- Other typo fixes: ([#4628](https://github.com/dotnet/machinelearning/pull/4628), [#4685](https://github.com/dotnet/machinelearning/pull/4685))
```

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+- Fix bug in WordBagEstimator when training on empty data ([#4696](https://github.com/dotnet/machinelearning/pull/4696))
+- Added Cancellation mechanism to Image Classification (through the experimental nuget) (fixes #4632) ([#4650](https://github.com/dotnet/machinelearning/pull/4650))
+-	Changed F1 score to return 0 instead of NaN when Precision + Recall is 0 ([#4674](https://github.com/dotnet/machinelearning/pull/4674))
+- TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
+- Changed onnx export to append a .onnx string to column names to support better mapping to ML.NET output column names ([#4734](https://github.com/dotnet/machinelearning/pull/4734))
+-	TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
+- ImageLoadingTransformer now checks the existence of input folder before training ([#4691]( https://github.com/dotnet/machinelearning/pull/4691))
+- Sweep Trimming of Whitespace in AutoML ([#3918](https://github.com/dotnet/machinelearning/pull/3918))
```
Comment (justinormont) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: This isn't in master; only in the AutoML branch. After testing it should be cherry picked to master. No need to mention here.

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md (hunk 10 above)
Comment (codemzs) [reply docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: thanks and in addition I would also
expand LDSVM acronym.

Comment (justinormont) [reply docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: "Local Deep SVM"

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
+# [ML.NET](http://dot.net/ml) 1.5.0-preview2
+
+## **New Features (IN-PREVIEW, please provide feedback)**
+- **TimeSeriesImputer** ([#4623](https://github.com/dotnet/machinelearning/pull/4623)) This data transformer can be used to impute missing rows in time series data.
+- **LDSVM Trainer** ([#4060](https://github.com/dotnet/machinelearning/pull/4060)) The "Local Deep SVM" usess trees as its SVM kernel to create a non-linear binary trainer. A sample can be found [here](https://github.com/dotnet/machinelearning/blob/c819d77e9250c68883713d5f1cd79b8971a11faf/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LdSvm.cs).
```
Comment (najeeb-kazmi) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: 'usess' -> 'uses'

File: docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+  
+## **Bug Fixes**
+- Fix for OneVersusAllTrainer ([#4698](https://github.com/dotnet/machinelearning/pull/4698))
+- Added support to run PFI on uncalibrated binary classification models ([#4587](https://github.com/dotnet/machinelearning/pull/4587))
+- Fix bug in WordBagEstimator when training on empty data ([#4696](https://github.com/dotnet/machinelearning/pull/4696))
+- Added Cancellation mechanism to Image Classification (through the experimental nuget) (fixes #4632) ([#4650](https://github.com/dotnet/machinelearning/pull/4650))
+-	Changed F1 score to return 0 instead of NaN when Precision + Recall is 0 ([#4674](https://github.com/dotnet/machinelearning/pull/4674))
+- TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
+- Changed onnx export to append a .onnx string to column names to support better mapping to ML.NET output column names ([#4734](https://github.com/dotnet/machinelearning/pull/4734))
+-	TextLoader, BinaryLoader and SvmLightLoader now check the existence of the input file before training ([#4665](https://github.com/dotnet/machinelearning/pull/4665))
```
Comment (eerhardt) [inline docs/release-notes/1.5.0-preview2/release-1.5.0-preview2.md]: This is duplicate of the line 2 above.
